debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -118,6 +125,8 @@ public SchemaBuilder schemaBuilder(Column column) {
             case PgOid.JSONB_OID:
             case PgOid.JSON:
                 return Json.builder();
+            case PgOid.HSTORE:
+                return SchemaBuilder.map(Schema.STRING_SCHEMA,Schema.STRING_SCHEMA);
```

Comment:
This should examine the HStoreHandlingMode to create the string or map schema.

Reply:
private SchemaBuilder hStoreSchema(){
        if (hStoreMode == HStoreHandlingMode.MAP){
            return SchemaBuilder.map(Schema.STRING_SCHEMA,Schema.STRING_SCHEMA);
        }
        return Json.builder();
    }
and then call this method in the HSTORE case .